### PR TITLE
[JENKINS-57797] Support Configuration-as-Code

### PR DESCRIPTION
**Pull request by alecharp**
[JENKINS-57797](https://issues.jenkins-ci.org/browse/JENKINS-57797)

This PR aims to validate the support of configuration-as-code for this plugin.

By default, the configuration is supported, but under the section `sSHLauncher`, this is why I added the `@Symbol` on the `SSHLauncher` descriptor.

### CODE REVIEW

File: src/main/java/hudson/plugins/sshslaves/SSHLauncher.java [hunk 1]
```diff
@@ -1223,6 +1224,7 @@ public String getWorkDirParam(@NonNull String workingDirectory){
     }
 
     @Extension
+    @Symbol({"sSHLauncher", "ssh"})
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/sshslaves/SSHLauncher.java]: With such ordering `sSHLauncher` will be a default alias for export. I suggest keeping it only for compatibility purposes

```suggestion
    @Symbol({"ssh", "sSHLauncher"})
```

**Comment by alecharp** [comment]
it's missing the test about the round-trip export/import of the configuration.

File: src/test/java/hudson/plugins/sshslaves/SSHLauncherCasCRoundTripTest.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    @Override
+    protected String stringInLogExpected() {
+        return "hudson.slaves.DumbSlave.name = this-ssh-agent";
+    }
+
+    @Override
+    protected String configResource() {
+        return "SSHCasCConfig.yml";
```
Comment (MRamonLeon) [inline src/test/java/hudson/plugins/sshslaves/SSHLauncherCasCRoundTripTest.java]: FTR: If you rename the yml file to _configuration-as-code.yaml_ you don't need to override that.

Comment (alecharp) [reply src/test/java/hudson/plugins/sshslaves/SSHLauncherCasCRoundTripTest.java]: I have two files. I prefer each on to have a explicit name.

**Comment by batmat** [comment]
@kuisathaverat I think this is ready for a maintainer review :). Could you please have a look, and possibly merge & release this? 

Gracias!

**Comment by kuisathaverat** [comment]
@batmat @alecharp I'll try to release it this weekend

**Comment by MRamonLeon** [comment]
:+1: We still love you @kuisathaverat

File: pom.xml [hunk 3]
```diff
@@ -27,6 +27,8 @@
   <properties>
     <jenkins.version>2.150.1</jenkins.version>
     <java.level>8</java.level>
+    <!-- TODO fix once released -->
+    <configuration-as-code.version>1.20-20190604.105601-2</configuration-as-code.version>
```
Comment (batmat) [inline pom.xml]: @kuisathaverat wooops, there's actually an incrementals in use here. I don't think we should release in this state.

Comment (kuisathaverat) [reply pom.xml]: it is a test dependency, I guess if I try with the latest version and the test pass is ok to release changing the dependency to the latest version.

Comment (MRamonLeon) [reply pom.xml]: Waiting for the release today, most likely. CC: @oleg-nenashev @casz

**Pull request merged**